apps/rush-lib/src/utilities/Utilities.ts

```diff
@@ -548,23 +622,42 @@ export class Utilities {
         continue;
       }
 
-      environment[key] = initialEnvironment[key];
+      environment[key] = options.initialEnvironment[key];
     }
 
-     // When NPM invokes a lifecycle script, it sets an environment variable INIT_CWD that remembers
-     // the directory that NPM started in.  This allows naive scripts to change their current working directory
-     // and invoke NPM operations, while still be able to find a local .npmrc file.  Although Rush recommends
-     // for toolchain scripts to be professionally written (versus brittle stuff like
-     // "cd ./lib && npm run tsc && cd .."), we support INIT_CWD for compatibility.
-     //
-     // More about this feature: https://github.com/npm/npm/pull/12356
-     if (initCwd) {
-       environment['INIT_CWD'] = initCwd; // tslint:disable-line:no-string-literal
-     }
+    // When NPM invokes a lifecycle script, it sets an environment variable INIT_CWD that remembers
+    // the directory that NPM started in.  This allows naive scripts to change their current working directory
+    // and invoke NPM operations, while still be able to find a local .npmrc file.  Although Rush recommends
+    // for toolchain scripts to be professionally written (versus brittle stuff like
+    // "cd ./lib && npm run tsc && cd .."), we support INIT_CWD for compatibility.
+    //
+    // More about this feature: https://github.com/npm/npm/pull/12356
+    if (options.initCwd) {
+      environment['INIT_CWD'] = options.initCwd; // tslint:disable-line:no-string-literal
+    }
+
+    if (options.pathOptions) {
+      if (options.pathOptions.includeRepoBin && options.pathOptions.commonTempFolder) {
```

Comment:
If both are specified, the project folder should take precedence over the common/temp folder.  So reverse these two if-statements.

Reply:
If PATH is `/A:/B:/C`, `/A` has precedence over `/B`, right? So the order of the if statements should be correct

Reply:
> If PATH is `/A:/B:/C`, `/A` has precedence over `/B`, right? 

Correct.

> So the order of the if statements should be correct

Incorrect.  The code should add `projectRoot` before `commonTempFolder`

Reply:
>Incorrect. The code should add projectRoot before commonTempFolder

The code as written will prepend the `commonTempFolder` and then prepend `projectRoot`.
1. `PATH` starts out as `/X`
2. `PATH` becomes `/commonTemp:/X`
3. `PATH` becomes `/projectFolder:/commonTemp:/X`

Reply:
Ohh I get it.  `addNodeModulesBinToPath` inserts at the start of the list.

Can you rename this function to something like `prependNodeModulesBinToPath`?  And maybe write some docs?